authors/apps/articles/views.py

```diff
@@ -365,3 +365,47 @@ def get(self, request, *args, **kwargs):
             },
                 status=status.HTTP_404_NOT_FOUND)
         return Response(serializer.data, status.HTTP_200_OK)
+
+
+class ReportArticleView(generics.CreateAPIView):
+    """
+    View for user to report article
+    """
+    permission_classes = (IsAuthenticated,)
+    serializer_class = ReportSerializer
+    queryset = ReportArticle.objects.all()
+
+    def get_object(self, *args, **kwargs):
+        slug = self.kwargs.get('slug')
+        return get_object_or_404(Article, slug=slug)
+
+    def get_queryset(self):
+        article = self.get_object()
+        reported_article = self.queryset.filter(
+            reported_by=self.request.user, article=article).first()
+        return reported_article
+
+    def post(self, request, *args, **kwargs):
+        report = self.request.data
+        article = self.get_object()
+        user = self.request.user
+        reported_article = self.get_queryset()
+        if reported_article:
```

Comment:
@VivianDoreen No need to create vatriable reoprted_article
```suggestion
        if self.get_queryset():
```

Reply:
@Opio-Emmanuel-Omona ,Am done with the changes. Kindly review to confirm